packages/ui-causal-graph-editor/src/shared/graph-layout/fcose-layout.tsx

```diff
@@ -156,10 +156,10 @@ class FcoseLayoutBuilder extends GraphLayoutBuilder<FcoseLayout> implements Tier
 
 /**
  * Based on a node attribute checks if the path is in the attribute or in extras, if not found returns undefined
- * @param path path to the attribute
  * @param attributes node object or a sub attribute of the node object
+ * @param path path to the attribute
  *  */
-function getPathInNodeAttribute(path: string, attributes: any): any {
+function getPathInNodeAttribute(attributes: Record<string, any>, path: string): any {
```

Comment:
I might be missing something but this function might need to have a guard for `attributes` being undefined, e.g. if you'd  provide path as `foo.bar` and foo doesn't exist, it would return undefined an then the next reduce call would attempt to access a property on undefined which throws an Error.
Can we add few quick unit tests for these to make sure this works correctly?

Reply:
Added an error for this, was adding the tests just finished them now and added oneto cover this scenario as well